corehq/apps/locations/tasks.py

```diff
@@ -124,6 +125,17 @@ def update_users_at_locations(domain, location_ids, supply_point_ids, ancestor_i
     update_fixture_status_for_users(user_ids, UserLookupTableType.LOCATION)
 
 
+@task
+def delete_locations_related_rules(location_ids):
+    for location_definition in LocationFilterDefinition.objects.filter(location_id__in=location_ids):
+        for criteria in location_definition.caserulecriteria_set.all():
+            rule = criteria.rule
+            rule.delete_criteria()
+            rule.delete_actions()
+            rule.delete()
```

Comment:
@esoergel 
> Don't rules commonly have multiple criteria?

Are you asking whether rules share criteria? If so, the answer is no. You can see [here](https://github.com/dimagi/commcare-hq/blob/db36c7d617756ea5b27e1c360769ead347bd5e4b/corehq/apps/data_interfaces/forms.py#L657) when the criteria is [added to the rule](https://github.com/dimagi/commcare-hq/blob/c8eb0c252525806f28f158cf23b7c08e0f7e562c/corehq/apps/data_interfaces/views.py#L1020), the existing criteria is removed completely and recreated, meaning that the criteria on any rule is only used on that rule and not shared with any other rule. 

It may also be worth mentioning that any rule can only have exactly one location filter (at least at the moment; OR logic is not implemented for dedupe yet), so when we're iterating over the location_definitions and looping over their respective criteria, the rule returned when we call `criteria.rule` will always be unique for every iteration.

As for the tests, I'll add more verbose testing :+1:

Reply:
I was asking whether there can be more than one criteria object pointing to the same rule.

> so when we're iterating over the location_definitions and looping over their respective criteria, the rule returned when we call criteria.rule will always be unique for every iteration.

Perfect, this is what I was wondering about - that sounds fine then.